drivers/gpio/gpio_mcux.c

```diff
@@ -261,6 +261,25 @@ static void gpio_mcux_port_isr(const struct device *dev)
 	gpio_fire_callbacks(&data->callbacks, dev, int_status);
 }
 
+#ifdef CONFIG_GPIO_GET_DIRECTION
+static int gpio_mcux_port_get_direction_raw(const struct device *dev,
+						 gpio_port_pins_t *inputs,
+						 gpio_port_pins_t *outputs)
+{
+	const struct gpio_mcux_config *config = dev->config;
+	GPIO_Type *gpio_base = config->gpio_base;
+
+	if (inputs != NULL) {
+		*inputs = ~gpio_base->PDDR;
+	}
+
+	if (inputs != NULL) {
+		*outputs = gpio_base->PDDR;
+	}
```

Comment:
This doesn't look right? inputs vs. outputs?

Reply:
Fixed!